keanu-project/src/test/java/io/improbable/keanu/vertices/dbl/probabilistic/ParetoVertexTest.java

```diff
@@ -0,0 +1,235 @@
+        scaleTensor.setValue(1.5);
+
+        Supplier<ParetoVertex> vertexSupplier = () -> new ParetoVertex(locationTensor, scaleTensor);
+
+        ProbabilisticDoubleTensorContract.matchesKnownDerivativeLogDensityOfVector(vector, vertexSupplier);
+    }
+
+    @Test
+    public void isTreatedAsConstantWhenObserved() {
+        UniformVertex xm = new UniformVertex(0.0, 1.0);
+        xm.setAndCascade(Nd4jDoubleTensor.scalar(1.0));
+        ParetoVertex vertexUnderTest = new ParetoVertex(xm, 3.0);
+        vertexUnderTest.setAndCascade(1.0);
+        ProbabilisticDoubleTensorContract.isTreatedAsConstantWhenObserved(vertexUnderTest);
+        ProbabilisticDoubleTensorContract.hasNoGradientWithRespectToItsValueWhenObserved(vertexUnderTest);
+    }
+
+    @Test
+    public void dLogProbMatchesFiniteDifferenceCalculationFordPdXm() {
+        UniformVertex xm = new UniformVertex(0.0, 1.0);
+        ParetoVertex pareto = new ParetoVertex(xm, 3.0);
+
+        DoubleTensor vertexStartValue = Nd4jDoubleTensor.scalar(1.5);
+        DoubleTensor vertexEndValue = Nd4jDoubleTensor.scalar(5.0);
+        double vertexInc = 0.1;
+
+        moveAlongDistributionAndTestGradientOnARangeOfHyperParameterValues(
+            Nd4jDoubleTensor.scalar(0.1),
+            Nd4jDoubleTensor.scalar(1.0),
+            0.1,
+            xm,
+            pareto,
+            vertexStartValue,
+            vertexEndValue,
+            vertexInc,
+            1e-5
+        );
+    }
+
+    @Test
+    public void dLogProbMatchesFiniteDifferenceCalculationFordPdAlpha() {
+        UniformVertex alpha = new UniformVertex(0.1, 5.0);
+        ParetoVertex pareto = new ParetoVertex(1.0, alpha);
+
+        DoubleTensor vertexStartValue = Nd4jDoubleTensor.scalar(3.0);
+        DoubleTensor vertexEndValue = Nd4jDoubleTensor.scalar(5.0);
+        double vertexInc = 0.1;
+
+        moveAlongDistributionAndTestGradientOnARangeOfHyperParameterValues(
+            Nd4jDoubleTensor.scalar(0.1),
+            Nd4jDoubleTensor.scalar(5.0),
+            0.1,
+            alpha,
+            pareto,
+            vertexStartValue,
+            vertexEndValue,
+            vertexInc,
+            0.01
+        );
+    }
+
+    @Test
+    public void sampleMatchesLogProb() {
+        int sampleCount = 1000000;
+        ParetoVertex vertex = new ParetoVertex(new int[]{sampleCount, 1}, 1.0, 3.0);
+
+        double from = 1.0;
+        double to = 2.5;
+        double bucketSize = 0.01;
+
+        ProbabilisticDoubleTensorContract.sampleMethodMatchesLogProbMethod(vertex, from, to, bucketSize, 5e-2,
+            random);
+    }
+
+    @Test
+    public void inferHyperParamsFromSamples() {
+        double trueXm = 5.0;
+
+        /*
+         * Note, this value is set low as the Gradient Optimizer seems to struggle with values greater than this - we
+         * should revisit to see if the Optimizer can be Optimzed to deal better with this case.
```

Comment:
`we should revisit this with a more configurable Optimiser or a different Optimiser entirely!`